Reduce deprecated usages

WojciechSova opened the pull request.

piotrrzysko commented:
Have you checked if `RestTemplateHermesSender` is no longer used by `hermes-client` users? I mean our internal users, as it is probably impossible to determine how many (if any) external ones use it.

WojciechSova commented on integration/build.gradle:
```diff
@@ -25,7 +25,7 @@ dependencies {
     testCompile group: 'org.apache.commons', name: 'commons-lang3', version: '3.0'
     testCompile group: 'commons-beanutils', name: 'commons-beanutils', version: '1.9.4'
     testCompile group: 'org.apache.curator', name: 'curator-test', version: versions.curator
-    testCompile group: 'org.springframework', name: 'spring-web', version: versions.spring_web
```
It's included in [spring-webflux](https://github.com/spring-projects/spring-framework/blob/main/spring-webflux/spring-webflux.gradle#L8)

WojciechSova commented on integration/src/integration/java/pl/allegro/tech/hermes/integration/management/OfflineRetransmissionManagementTest.java:
```diff
@@ -78,8 +79,8 @@ public void shouldReturnClientErrorWhenRequestingRetransmissionWithEmptyData() {
         assertThat(response).containsMessages(
-                "sourceTopic may not be empty",
-                "targetTopic may not be empty",
+                "sourceTopic must not be empty",
```
The change of @NotEmpty validator from hibernate to javax changed the message

WojciechSova commented:
> Have you checked if `RestTemplateHermesSender` is no longer used by `hermes-client` users? I mean our internal users, as it is probably impossible to determine how many (if any) external ones use it.

Yes, you are right, I forgot about that. I added it back and also added the Deprecated annotation.

piotrrzysko commented on integration/src/integration/java/pl/allegro/tech/hermes/integration/AsyncPublisherAutoRetryTest.java:
```diff
@@ -59,7 +59,7 @@ public void shouldNotRetryIfMessageWasPublished() {
 
     private HermesClient retryingHermesClient(int retries) {
-         return HermesClientBuilder.hermesClient(new RestTemplateHermesSender(new AsyncRestTemplate()))
+         return HermesClientBuilder.hermesClient(new WebClientHermesSender(WebClient.create()))
```
I'm not sure why we're changing this. I'd rather vote for adding separate tests for `WebClientHermesSender`. 
Even though `RestTemplateHermesSender` is deprecated, we're still maintaining it, so it would be nice to have tests to see if it works properly.

szczygiel-m commented on integration/src/integration/java/pl/allegro/tech/hermes/integration/AsyncPublisherAutoRetryTest.java:
```diff
@@ -40,10 +43,11 @@ public void before() {
 
     @Test
-    public void shouldRetryPublishingMessageEnoughTimesWhenServerReturns500() {
+    public void shouldRetryPublishingMessageEnoughTimesWhenServerReturns500UsingRestTemplate() {
```
Cool that we are testing deprecated client until it is fully removed 👍

pitagoras3 replied:
Nit: as this test uses `AsyncRestTemplate`, maybe it should be named `shouldRetryPublishingMessageEnoughTimesWhenServerReturns500UsingAsyncRestTemplate`?

pitagoras3 commented on integration/src/integration/java/pl/allegro/tech/hermes/integration/AsyncPublisherAutoRetryTest.java:
```diff
@@ -53,8 +57,9 @@ public void shouldRetryPublishingMessageEnoughTimesWhenServerReturns500() {
     @Test
-    public void shouldNotRetryIfMessageWasPublished() {
-        HermesClient hermesClient = retryingHermesClient(10);
+    public void shouldNotRetryIfMessageWasPublishedUsingRestTemplate() {
```
Nit: as this test uses `AsyncRestTemplate`, maybe it should be named `shouldNotRetryIfMessageWasPublishedUsingAsyncRestTemplate`?
